### ISSUE TITLE
Support for 1D array texture views

### ISSUE THREAD

**Issue by kvark** [gpuweb/gpuweb#699]
Spawned from https://github.com/gpuweb/gpuweb/pull/613#issuecomment-613035778
We need a bit of an investigation, potentially followed by a way to provide the layer count.

**Comment by kainino0x** [comment]
Is there something to discuss in the meeting or do we need the investigation first?

**Comment by kvark** [comment]
Right, I mentioned that in the editor meeting. This needs an investigation first.

**Comment by kainino0x** [comment]
This is supposed to be about 1d-array texture views, right? Title doesn't say it.

**Comment by kainino0x** [comment]
Corentin found #79 which points to this table:
https://developer.apple.com/documentation/metal/mtltexture/1515409-newtextureviewwithpixelformat#discussion
which seems to say texture views can't be created from 1DArray at all.

**Comment by kvark** [comment]
I suppose that's good news. One less confusion about where the depth goes :)

**Comment by kainino0x** [comment]
We were wondering about whether it's possible to emulate them with 2D array textures of height 1. Should we consider it?

**Comment by kvark** [comment]
I think we can leave this opportunity hanging while listening to ISV feedback.

**Comment by kainino0x** [comment]
SGTM

**Comment by kvark** [comment]
> Corentin found #79 which points to this table:
> https://developer.apple.com/documentation/metal/mtltexture/1515409-newtextureviewwithpixelformat#discussion
> which seems to say texture views can't be created from 1DArray at all.

Hmm, technically speaking Metal doesn't differentiate between texture and views. So in Metal we could use the original texture as a view, but it would only work if it covers the whole texture, i.e. the "default view".

**Comment by kainino0x** [comment]
You're right, I forgot about that somehow.
